lib/parse-font.js

```diff
@@ -1,64 +1,101 @@
+  , units = 'px|pt|pc|in|cm|mm|%|em|ex|ch|rem|q'
+  , string = '\'([^\']+)\'|"([^"]+)"|[\\w-]+'
+
+// [ [ <‘font-style’> || <font-variant-css21> || <‘font-weight’> || <‘font-stretch’> ]?
+//    <‘font-size’> [ / <‘line-height’> ]? <‘font-family’> ]
+// https://drafts.csswg.org/css-fonts-3/#font-prop
+const weightRe = new RegExp(`(${weights}) +`, 'i')
+const styleRe = new RegExp(`(${styles}) +`, 'i')
+const variantRe = new RegExp(`(${variants}) +`, 'i')
+const stretchRe = new RegExp(`(${stretches}) +`, 'i')
+const sizeFamilyRe = new RegExp(
+  '([\\d\\.]+)(' + units + ') *'
+  + '((?:' + string + ')( *, *(?:' + string + '))*)')
 
 /**
- * Cache color string RGBA values.
+ * Cache font parsing.
  */
 
 const cache = {}
 
+const defaultHeight = 16 // pt, common browser default
+
 /**
  * Parse font `str`.
  *
  * @param {String} str
- * @return {Object}
+ * @return {Object} Parsed font. `size` is in device units. `unit` is the unit
+ *   appearing in the input string.
  * @api private
  */
 
 module.exports = function (str) {
-  let parsedFont
-
-  // Try to parse the font string using parse-css-font.
-  // It will throw an exception if it fails.
-  try {
-    parsedFont = parseCssFont(str)
-  } catch (_) {
-    // Invalid
-    return undefined
-  }
-
   // Cached
   if (cache[str]) return cache[str]
 
-  // Parse size into value and unit using units-css
-  var size = unitsCss.parse(parsedFont.size)
+  // Try for required properties first.
+  const sizeFamily = sizeFamilyRe.exec(str)
+  if (!sizeFamily) return // invalid
+
+  // Default values and required properties
+  const font = {
+    weight: 'normal',
+    style: 'normal',
+    stretch: 'normal',
+    variant: 'normal',
+    size: parseFloat(sizeFamily[1]),
+    unit: sizeFamily[2],
+    family: sizeFamily[3].replace(/["']/g, '').replace(/ *, */g, ',')
+  }
 
-  // TODO: dpi
-  // TODO: remaining unit conversion
-  switch (size.unit) {
+  // Optional, unordered properties.
+  let weight, style, variant, stretch
+  // Stop search at `sizeFamily.index`
+  let substr = str.substring(0, sizeFamily.index)
+  if ((weight = weightRe.exec(substr))) font.weight = weight[1]
+  if ((style = styleRe.exec(substr))) font.style = style[1]
+  if ((variant = variantRe.exec(substr))) font.variant = variant[1]
+  if ((stretch = stretchRe.exec(substr))) font.stretch = stretch[1]
+
+  // Convert to device units. (`font.unit` is the original unit)
+  // TODO: ch, ex
+  switch (font.unit) {
     case 'pt':
-      size.value /= 0.75
+      font.size /= 0.75
+      break
+    case 'pc':
+      font.size *= 16
       break
     case 'in':
-      size.value *= 96
+      font.size *= 96
+      break
+    case 'cm':
+      font.size *= 96.0 / 2.54
       break
     case 'mm':
-      size.value *= 96.0 / 25.4
+      font.size *= 96.0 / 25.4
       break
-    case 'cm':
-      size.value *= 96.0 / 2.54
+    case '%':
```

Comment:
So yeah, I don't think the test is right... the units do get sent to C++, but [aren't used](https://github.com/Automattic/node-canvas/blob/25b03f3e34bf25ca62f9069ebddc9b9335d9b6df/src/CanvasRenderingContext2d.cc#L1998). We probably don't need to send the units, just the normalized `px` value.

It looks like the current version would be doing percentages and `mm` wrong, I'll have to look at if the non-Pango build was ever right about that.